src/main/java/com/mitchej123/hodgepodge/asm/transformers/optifine/GLErrorLoggingTransformer.java

```diff
@@ -0,0 +1,51 @@
+package com.mitchej123.hodgepodge.asm.transformers.optifine;
+
+import static org.objectweb.asm.Opcodes.ASM5;
+
+import net.minecraft.launchwrapper.IClassTransformer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.InsnList;
+import org.objectweb.asm.tree.InsnNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+
+public class GLErrorLoggingTransformer implements IClassTransformer {
+
+    private static final Logger LOGGER = LogManager.getLogger("GLErrorLogging");
+
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] basicClass) {
+        if ("shadersmod.client.Shaders".equals(transformedName)) {
+            LOGGER.info("TRANSFORMING shadersmod.client.Shaders");
+            final ClassReader cr = new ClassReader(basicClass);
+            final ClassNode cn = new ClassNode(ASM5);
+            cr.accept(cn, 0);
+            for (MethodNode m : cn.methods) {
+                if ("checkGLError".equals(m.name)) {
+                    LOGGER.info("Removing GL Error Logging: " + m.desc);
+                    final InsnList i = m.instructions;
+                    i.clear();
+                    i.add(
+                            new MethodInsnNode(
+                                    Opcodes.INVOKESTATIC,
+                                    "org/lwjgl/opengl/GL11",
+                                    "glGetError",
+                                    "()I",
+                                    false));
+                    i.add(new InsnNode(Opcodes.IRETURN));
```

Comment:
you need to change max locals / max stack and probably remove some local variables from the table

Reply:
I have no idea how. Got any examples?

Reply:
Is it like this?
`final ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_MAXS);`

Reply:
that will recompute the MAX, but it won't clean the local variable table

Reply:
well how can I clean the local variable table? Got an example?

Reply:
this is a field in `MethodNode` you look inside the list and remove the variables that you deleted with your ASM
![image](https://github.com/GTNewHorizons/Hodgepodge/assets/57050655/7a825398-8371-4438-ab5c-a39173bf7de5)